src/sub_command/set.rs

```diff
@@ -901,13 +899,19 @@ edition = "2021"
         insert_new_msrv(&mut manifest, &METADATA_MSRV).unwrap();
 
         let output = manifest.to_string();
-        let new_manifest: CargoManifest = CargoManifestParser
-            .parse::<Document>(&output)
-            .unwrap()
-            .try_into()
+        let document = CargoManifestParser.parse::<Document>(&output).unwrap();
+
+        let msrv = document
+            .as_table()
+            .get("package")
+            .and_then(Item::as_table)
+            .and_then(|package| package.get("metadata"))
+            .and_then(Item::as_table_like)
+            .and_then(|metadata| metadata.get("msrv"))
+            .and_then(Item::as_str)
             .unwrap();
 
-        assert_eq!(new_manifest.minimum_rust_version().unwrap(), &METADATA_MSRV)
+        assert_eq!(msrv, &METADATA_MSRV.to_string());
```

Comment:
Just inlined the parsing toml code, as can't use cargo metadata here easily (need a file, not just a string)